LSP.sublime-settings

```diff
@@ -130,6 +130,30 @@
       "scopes": ["source.julia"],
       "syntaxes": ["Packages/Julia/Julia.sublime-syntax"],
       "settings": {"runlinter": true}
+    },
+    "polymer-ide":
+    {
+      "command": ["polymer-editor-service"],
+      "scopes": [
+        "text.html.basic",
+        "text.html",
+        "source.html",
+        "source.js",
+        "source.css"
+      ],
+      "syntaxes": [
+        "Packages/HTML/HTML.sublime-syntax",
+        "Packages/CSS/CSS.sublime-syntax",
+        "Packages/JavaScript/JavaScript.sublime-syntax",
+        "Packages/JavaScript/JSON.sublime-syntax"
```

Comment:
Does the language server provide services in documents of all these types (even JSON?)?

Reply:
Is this something where syntaxes and scopes should be different? I'm not sure how they're used by LSP.

Reply:
Sorry, I thought I'd merged this long ago :blush:

The scope vs. syntax is mostly an implementation detail:

LSP uses two listener Apis: EventListener (global) and ViewEventListener (per file)

Where possible, we use ViewEventListeners for views of the right syntax. This way we don't slow down the UI or cause unintended behaviour for irrelevant files.

The EventListeners are global, so there we can check the scope of the code around the cursor. Example, if sublime detects javascript embedded within a document with HTML syntax, LSP can provide services for that.